File: fe/whotapp/src/app/models/card.ts
import { Player } from "./player"

export interface Card {
    id: number,
    name: string,
    value: number,
    isSpecial: boolean,
    color: string, //TODO change to Enum do I really need enum
    shape: string //TODO change to Enum
}

export interface CardOptionRequest{
  color: string,
  shape: string
}

export interface PlayCardRequest{
  gameId : number,
  cardId: number,
  cardColor: string,
  cardShape: string
}

export interface GameStats {
  id: number,
  status: string,
  pick2Count: number,
  pick4Count: number,
  isTurnReversed: boolean,
  marketCount: number,
  lastPlayerId: number,
  currentPlayerId: number,
  lastPlayedCard: Card,
  players: Player[]
}

File: be/WhotGame.Core/Data/Models/Card.cs
using WhotGame.Core.Enums;

namespace WhotGame.Core.Data.Models
{
    public class Card
    {
        public int Id { get; set; }
        public CardColor? Color { get; set; }
        public CardShape? Shape { get; set; }
        public string Name { get; set; }
        public int Value { get; set; }
        public bool IsSpecial { get; set; }
    }
}
